app/addons/components/layouts.js

```diff
@@ -0,0 +1,94 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {NotificationCenterButton} from "../fauxton/notifications/notifications.react";
+import {ApiBarController} from './react-components.react';
+import {Breadcrumbs} from './header-breadcrumbs';
+import ComponentActions from "./actions";
+
+const ApiBarWrapper = ({docURL, endpoint}) => {
+  //TODO once all modules are using this remove actions and make them props
+  ComponentActions.updateAPIBar({
+    buttonVisible: true,
+    contentVisible: false,
+    endpoint,
+    docURL
+  });
+  return (
+    <div id="api-navbar">
+      <ApiBarController
+        buttonVisible={true}
+        contentVisible={false}
+      />
+  </div>
+  );
+};
+
+const CrumbsWrapper = ({crumbs}) => {
+  if (!crumbs || crumbs.length === 0) {
+    return null;
+  }
```

Comment:
not a blocker, just sharing my thoughts.

in general i'm not a big fan of giving the responsibility to an element to decide if they should render or not. i prefer to have as much  logic outside as possible:

```js
{crumbs ? <CrumbsWrapper crumbs={crumbs} /> : null}
```

i see that this way to handle it can be convenient, just wanted to share my thoughts. :)